cylc/uiserver/resolvers.py

```diff
@@ -397,10 +470,34 @@ async def service(
                 executor=self.executor
             )
 
-        else:
+        elif command == 'play':
             return await Services.play(
                 workflows,
                 kwargs,
                 self.workflows_mgr,
                 log=self.log
             )
+
+        raise NotImplementedError()
+
+    async def subscription_service(
+        self,
+        info: 'ResolveInfo',
+        _command: str,
+        workflows: List[Tokens],
+        task=None,
+        file=None
+    ):
+        if file:
+            # set the correct option for cat log from cylc flow
+            for key, value in JOB_LOG_OPTS.items():
+                if value == file:
+                    file = key
```

Comment:
From experimentation it looks like you can pass the raw file name to `cylc cat-log` so I think we can get away without this mapping?

```
$ cylc cat-log generic/run2//20200111T0000Z/fool/NN/ -f job.out
```